File: e-commerce-school-supply-web/src/types/product.ts
export interface Product {
  id?: string
  name: string
  description?: string
  mainCategory?: string
  subCategory?: string
  type?: string
  size?: string
  color?: string
  material?: string
  brandName?: string
  stockQuantity?: number
  price: number
  discount?: number | null
  // Prefer the new multi-image field; fall back to legacy single url when needed
  images?: string[]
  imageUrl?: string
  status?: 'In Stock' | 'Out of stock'
  rating?: number | 0
  averageRating: number
  reviewCount?: number
  imageURL?: string // For backward compatibility
}

File: e-commerce-school-supply-api/src/main/java/com/e_commerce/backend/entity/Product.java
package com.e_commerce.backend.entity;

import java.util.ArrayList;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;

@Data
@Document(collection = "products")
public class Product {
    @Id
    private String id; 
    private String name;
    private String description;
    private String mainCategory;
    private String subCategory;
    private String type;
    private String size;
    private String color;
    private String material;
    private String brandName;
    private Integer stockQuantity;
    private Double price;
    private Double discount;
    private List<String> images = new ArrayList<>();
    private String status;
    private Double averageRating;
    private Integer reviewCount;
}